src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -240,11 +255,19 @@ public void Set(int index, bool value)
         =========================================================================*/
         public void SetAll(bool value)
         {
-            int fillValue = value ? unchecked(((int)0xffffffff)) : 0;
-            int ints = GetArrayLength(m_length, BitsPerInt32);
-            for (int i = 0; i < ints; i++)
+            int ints = GetIntegerArrayLengthFromBitLength(m_length);
```

Comment:
Looks like size 10 is the break-even point for SetAll. Otherwise, for small lengths, this change causes 2x regression.

I will revert.

Alternatively, I could try something like this:
```C#
public void SetAll(bool value)
{
    int ints = GetIntegerArrayLengthFromBitLength(m_length);
    Span<sbyte> span = MemoryMarshal.Cast<int, sbyte>(m_array);
    if (value)
    {
        span.Fill(-1);
    }
    else
    {
        span.Clear();
                
    }

    _version++;
}
```